cloud-template/src/main/java/com/sequenceiq/cloudbreak/cloud/template/AbstractResourceConnector.java

```diff
@@ -180,9 +185,29 @@ public ExternalDatabaseStatus getDatabaseServerStatus(AuthenticatedContext authe
         context.addGroupResources(scalingGroup.getName(), groupResourceService.getGroupResources(variant, resources));
 
         //compute
+        diskReattachment(resources, scalingGroup, context);
         return computeResourceService.buildResourcesForUpscale(context, auth, stack, Collections.singletonList(scalingGroup));
     }
 
+    protected void diskReattachment(List<CloudResource> resources, Group scalingGroup, ResourceBuilderContext context) {
+        List<CloudResource> diskSets = resources.stream()
+                .filter(cloudResource -> scalingGroup.getName().equalsIgnoreCase(cloudResource.getGroup()))
+                .filter(cloudResource -> getDisResourceType().equals(cloudResource.getType()))
+                .filter(cloudResource -> StringUtils.isEmpty(cloudResource.getInstanceId()) || CommonStatus.DETACHED.equals(cloudResource.getStatus()))
+                .collect(Collectors.toList());
+        for (int i = 0; i < diskSets.size(); i++) {
+            CloudResource cloudResource = diskSets.get(i);
+            Optional.ofNullable(scalingGroup.getInstances().get(i))
```

Comment:
We cannot guaranteed, we should refactor this code part

Reply:
Okay. I vaguely remember some service (HBase perhaps) storing host-dependent data on HDFS (or local FS, cannot remember exactly), so that re-attaching the wrong volumeset to an instance will result in a conflict due to the instance FQDN and disk data mismatch.